tracer/src/Datadog.Trace/Configuration/TracerSettings.cs

```diff
@@ -190,7 +190,11 @@ internal TracerSettings(IConfigurationSource? source, IConfigurationTelemetry te
                                         converter: x => x switch
                                         {
                                             "v1" or "V1" => SchemaVersion.V1,
+                                            // if we enable RemoveClientServiceNames, we need to use V1 to get the service mapping working
+                                            "v0" or "V0" when RemoveClientServiceNamesEnabled => SchemaVersion.V1,
                                             "v0" or "V0" => SchemaVersion.V0,
+                                            // for invalid and RemoveClientServiceNames, we will also use V1
+                                            _ when RemoveClientServiceNamesEnabled => SchemaVersion.V1,
```

Comment:
I wonder if we should remove this, and move it into the `getDefaultVersion` instead? That way the incorrect format gets recorded in telemetry correctly as an invalid format, and we fallback to the "default" 🤔 

e.g.
```csharp
() => RemoveClientServiceNamesEnabled
     ? new DefaultResult<SchemaVersion>(SchemaVersion.V1, "V1")
     : new DefaultResult<SchemaVersion>(SchemaVersion.V0, "V0")
```

This _should_ still pass the tests I think, but give clearer telemetry 🙂

Reply:
Thanks swapped to that